### PR TITLE
ebpf: Add documentation for `HashMap`

### PR DESCRIPTION
**Pull request by vadorovsky**

### CODE REVIEW

File: bpf/aya-bpf/src/maps/hash_map.rs [hunk 1]
```diff
@@ -47,38 +98,158 @@ impl<K, V> HashMap<K, V> {
...
+    /// # Safety
+    ///
+    /// Unless the map flag `BPF_F_NO_PREALLOC` is used, the kernel does not
+    /// make guarantee on the atomicity of [`HashMap::insert`] or
+    /// [`HashMap::remove`], and any element removed from the map might get
+    /// aliased by another element in the map, causing garbage to be read, or
+    /// corruption in case of writes.
+    ///
+    /// This method is safe, but it returns a raw pointer and deferefencing it
+    /// is an unsafe operation.
```
Comment (vadorovsky) [inline bpf/aya-bpf/src/maps/hash_map.rs:156]: I'm struggling to come up with any example which makes sense here.

File: bpf/aya-bpf/src/maps/mod.rs [hunk 2]
```diff
@@ -1,3 +1,32 @@
+//! Data structures used to setup and share data with eBPF programs.
+//!
+//! The eBPF platform provides data structures - maps in eBPF speak - that are
+//! used to store data and share them with user-space.
```
Comment (astoycos) [inline bpf/aya-bpf/src/maps/mod.rs:4]:
```suggestion
//! used to store and share data with user-space programs.
```

File: bpf/aya-bpf/src/maps/mod.rs [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
+//!
+//! # Examples
+//!
+//! XDP program using a [`HashMap<u32, u32>`](HashMap) to block traffic from
+//! IP addresses defined in it:
```
Comment (astoycos) [inline bpf/aya-bpf/src/maps/mod.rs:9]:
```suggestion
//! XDP program using a [`HashMap<u32, u32>`](HashMap) to block traffic from
//! certain IP addresses:
```

**Review by astoycos — COMMENTED** [review]
Just some small nit's / questions

**Pull request closed without merging**